File: mvc-architecture/mvc-arch/src/main/java/org/example/WebApplicationServer.java
package org.example;

import org.apache.catalina.LifecycleException;
import org.apache.catalina.startup.Tomcat;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.io.File;

public class WebApplicationServer {
    private static final Logger log = LoggerFactory.getLogger(WebApplicationServer.class);

    public static void main(String[] args) throws LifecycleException {
        String webappDirLocation = "webapps/";

        Tomcat tomcat = new Tomcat();
        tomcat.setPort(8080);
        tomcat.addWebapp("/", new File(webappDirLocation).getAbsolutePath());

        log.info("Configuring app with basedir : {}", new File("./" + webappDirLocation).getAbsolutePath());

        tomcat.start();
        tomcat.getServer().await();
    }
}

File: mvc-architecture/refactor-to-kotlin/src/main/kotlin/org/example/WebApplicationServer.kt
package org.example

import org.apache.catalina.startup.Tomcat
import org.slf4j.LoggerFactory
import java.io.File

class WebApplicationServer

fun main(args: Array<String>) {
    val log = LoggerFactory.getLogger(WebApplicationServer::class.java)
    val webappDirLocation = "webapps/"
    val tomcat = Tomcat()

    tomcat.setPort(8080)
    tomcat.addWebapp("/", File(webappDirLocation).absolutePath)

    log.debug("[WebApplicationServer] tomcat started on port 8080")

    tomcat.start()
    tomcat.server.await()
}